homeassistant/components/coolmaster/climate.py

```diff
@@ -151,6 +171,12 @@ async def async_set_fan_mode(self, fan_mode: str) -> None:
         self._unit = await self._unit.set_fan_speed(fan_mode)
         self.async_write_ha_state()
 
+    async def async_set_swing_mode(self, swing_mode: str) -> None:
+        """Set new swing mode."""
+        _LOGGER.debug("Setting swing mode of %s to %s", self.unique_id, swing_mode)
+        self._unit = await self._unit.set_swing(swing_mode)
```

Comment:
Is the upstream package returning the unit itself as response `set_swing`?
Seems a bit strange and what happens if the call fails?

Reply:
Yes, the upstream package always returns a new unit object (these are immutable), as a result of every mutating function. If a call fails, an exception will be raised.

This is the pattern used throughout this integration. Happy to have a discussion about it if you find it strange, but I don't think it should block @amitfin's PR - it's simply following the existing pattern.

Reply:
Then I think we should at least handle the exception to raise HomeAssistantError to give the user a proper response if the command did fail for some reason.

Reply:
Done

Reply:
https://user-images.githubusercontent.com/19599059/207962361-a6a9b79b-5bce-482c-abdf-486dac6ec793.mov